File: brand/naviar/.claude/skills/run-naviar/driver.mjs
#!/usr/bin/env node
/**
 * run-naviar/driver.mjs
 * NAVIAR brand contact sheet sürücüsü.
 * Kullanım:
 *   node driver.mjs shot <outdir>        → viewport ekran görüntüsü
 *   node driver.mjs full <outdir>        → tam sayfa görüntüsü
 *   node driver.mjs studies <outdir>     → her çalışmanın ayrı görüntüsü (S1–S10)
 *   node driver.mjs smoke <outdir>       → hızlı doğrulama (çıkış kodu 0/1)
 *
 * Yollar brand/naviar/ kökünden.
 */
import { createServer } from 'http';
import { createReadStream, existsSync, mkdirSync, readdirSync } from 'fs';
import { extname, join, resolve } from 'path';
import { createRequire } from 'module';
import { fileURLToPath } from 'url';

const __dir = fileURLToPath(new URL('.', import.meta.url));
// Serve the brand/naviar directory (2 levels up from skill dir)
const SERVE_ROOT = resolve(__dir, '../../../');
const PORT = 8091;

const MIME = {
  '.html': 'text/html', '.css': 'text/css', '.js': 'application/javascript',
  '.svg': 'image/svg+xml', '.png': 'image/png', '.jpg': 'image/jpeg',
  '.json': 'application/json', '.ico': 'image/x-icon',
};

function startServer() {
  return new Promise((ok, fail) => {
    const s = createServer((req, res) => {
      const file = join(SERVE_ROOT, req.url === '/' ? 'index.html' : req.url);
      if (!existsSync(file)) { res.writeHead(404); res.end(); return; }
      const mime = MIME[extname(file)] || 'application/octet-stream';
      res.writeHead(200, { 'Content-Type': mime });
      createReadStream(file).pipe(res);
    });
    s.listen(PORT, () => ok(s));
    s.on('error', fail);
    setTimeout(() => fail(new Error('server timeout')), 5000);
  });
}

async function withBrowser(fn) {
  const req = createRequire('/opt/node22/lib/node_modules/');
  const { chromium } = req('playwright');
  const browser = await chromium.launch({
    executablePath: '/opt/pw-browsers/chromium',
    args: ['--no-sandbox', '--disable-gpu'],
  });
  try { return await fn(browser); } finally { await browser.close(); }
}

const [cmd, outArg] = process.argv.slice(2);
const outDir = outArg || '/tmp/naviar-run';
mkdirSync(outDir, { recursive: true });

const server = await startServer();

try {
  if (cmd === 'shot' || !cmd) {
    await withBrowser(async browser => {
      const page = await browser.newPage({ viewport: { width: 1280, height: 900 } });
      await page.goto(`http://localhost:${PORT}/index.html`, { waitUntil: 'networkidle', timeout: 20000 });
      await page.waitForTimeout(800);
      const out = join(outDir, 'naviar-viewport.png');
      await page.screenshot({ path: out });
      console.log('SHOT:', out);
    });

  } else if (cmd === 'full') {
    await withBrowser(async browser => {
      const page = await browser.newPage({ viewport: { width: 1280, height: 900 } });
      await page.goto(`http://localhost:${PORT}/index.html`, { waitUntil: 'networkidle', timeout: 20000 });
      await page.waitForTimeout(800);
      const out = join(outDir, 'naviar-full.png');
      await page.screenshot({ path: out, fullPage: true });
      console.log('FULL:', out);
    });

  } else if (cmd === 'studies') {
    // Screenshot each study card individually by navigating to the SVG files
    const studiesDir = join(SERVE_ROOT, 'studies');
    const svgs = readdirSync(studiesDir).filter(f => f.endsWith('.svg')).sort();
    await withBrowser(async browser => {
      for (const svg of svgs) {
        const page = await browser.newPage({ viewport: { width: 800, height: 850 } });
        await page.goto(`http://localhost:${PORT}/studies/${svg}`, { waitUntil: 'networkidle', timeout: 15000 });
        const name = svg.replace('.svg', '');
        const out = join(outDir, `${name}.png`);
        await page.screenshot({ path: out });
        await page.close();
        console.log('STUDY:', out);
      }
    });

  } else if (cmd === 'smoke') {
    let pass = true;
    await withBrowser(async browser => {
      const page = await browser.newPage({ viewport: { width: 1280, height: 900 } });
      page.on('pageerror', e => { console.error('PAGE ERROR:', e.message); pass = false; });

      await page.goto(`http://localhost:${PORT}/index.html`, { waitUntil: 'networkidle', timeout: 20000 });

      // 1. Title check
      const title = await page.title();
      if (!title.includes('NAVIAR')) { console.error('❌ title missing NAVIAR'); pass = false; }
      else console.log('✅ title:', title);

      // 2. SVG studies present
      const svgCount = await page.evaluate(() =>
        document.querySelectorAll('img[src*="studies/"], object[data*="studies/"], .study-card').length
      );
      // Also check inline SVGs or iframes
      const studyCards = await page.evaluate(() => {
        const cards = document.querySelectorAll('[class*="study"],[id*="study"],[class*="card"]');
        return cards.length;
      });
      if (svgCount === 0 && studyCards === 0) {
        // Just check that some SVG content is in the page
        const hasSvg = await page.evaluate(() => !!document.querySelector('svg,img'));
        if (!hasSvg) { console.error('❌ no studies/images found'); pass = false; }
        else console.log('✅ content present (SVG/img detected)');
      } else {
        console.log(`✅ study elements: ${Math.max(svgCount, studyCards)}`);
      }

      // 3. Screenshot
      await page.screenshot({ path: join(outDir, 'smoke.png') });
      console.log('✅ screenshot saved');
    });

    console.log(pass ? 'SMOKE: PASS' : 'SMOKE: FAIL');
    process.exitCode = pass ? 0 : 1;
  } else {
    console.error('Bilinmeyen komut:', cmd);
    console.error('Kullanım: driver.mjs [shot|full|studies|smoke] <outdir>');
    process.exitCode = 1;
  }
} finally {
  server.close();
}

File: .claude/skills/run-brand/driver.py
#!/usr/bin/env python3
"""
run-brand driver — screenshot brand contact sheets via Playwright.

Usage:
  python3 .claude/skills/run-brand/driver.py smoke [outdir]
  python3 .claude/skills/run-brand/driver.py shot <brand> [light|dark] [outdir]
  python3 .claude/skills/run-brand/driver.py build [brand]

brands: coddan, cobbani, naviar, all
"""

import sys
import os
import subprocess
from pathlib import Path

REPO = Path(__file__).parent.parent.parent.parent
BRANDS = ["coddan", "cobbani", "naviar"]
CHROMIUM = "/opt/pw-browsers/chromium"


def get_page():
    from playwright.sync_api import sync_playwright
    return sync_playwright().__enter__()


def shot(brand: str, theme: str, outdir: Path) -> Path:
    """Screenshot one brand contact sheet, return path."""
    from playwright.sync_api import sync_playwright
    index = REPO / "brand" / brand / "index.html"
    if not index.exists():
        raise FileNotFoundError(f"No contact sheet at {index}")
    outdir.mkdir(parents=True, exist_ok=True)
    dest = outdir / f"{brand}-{theme}.png"
    with sync_playwright() as p:
        browser = p.chromium.launch(executable_path=CHROMIUM)
        page = browser.new_page(viewport={"width": 1440, "height": 900})
        if theme == "dark":
            page.emulate_media(color_scheme="dark")
        page.goto(f"file://{index}", wait_until="networkidle")
        errors = []
        page.on("console", lambda m: errors.append(m) if m.type == "error" else None)
        page.screenshot(path=str(dest))
        browser.close()
        if errors:
            print(f"  ⚠ console errors: {[e.text for e in errors]}")
    return dest


def cmd_smoke(outdir: Path):
    print("SMOKE: brand contact sheets")
    ok = True
    for brand in BRANDS:
        for theme in ["light", "dark"]:
            try:
                dest = shot(brand, theme, outdir)
                size = os.path.getsize(dest)
                if size < 10_000:
                    print(f"  ✗ {brand}/{theme}: suspiciously small ({size} B)")
                    ok = False
                else:
                    print(f"  ✓ {brand}/{theme}: {size:,} B → {dest}")
            except Exception as e:
                print(f"  ✗ {brand}/{theme}: {e}")
                ok = False
    if ok:
        print("SMOKE: PASS")
    else:
        print("SMOKE: FAIL")
        sys.exit(1)


def cmd_shot(brand: str, theme: str, outdir: Path):
    if brand == "all":
        for b in BRANDS:
            dest = shot(b, theme, outdir)
            print(f"  → {dest}")
    else:
        dest = shot(brand, theme, outdir)
        print(f"  → {dest}")


def cmd_build(brand: str):
    """Regenerate SVG/PNG assets via brand-specific build.py or global marka-uret.py."""
    if brand == "all" or brand is None:
        # Global QBLOGG brand assets
        script = REPO / "scripts" / "marka-uret.py"
        subprocess.run(["python3", str(script)], cwd=REPO, check=True)
        # Individual brand directories
        for b in BRANDS:
            build = REPO / "brand" / b / "build.py"
            if build.exists():
                print(f"\n── {b} ──")
                subprocess.run(["python3", str(build)], cwd=REPO / "brand" / b, check=True)
    else:
        build = REPO / "brand" / brand / "build.py"
        if build.exists():
            subprocess.run(["python3", str(build)], cwd=REPO / "brand" / brand, check=True)
        else:
            print(f"No build.py for {brand}; run scripts/marka-uret.py for QBLOGG assets")
            sys.exit(1)


if __name__ == "__main__":
    args = sys.argv[1:]
    if not args:
        print(__doc__)
        sys.exit(0)

    cmd = args[0]

    if cmd == "smoke":
        outdir = Path(args[1]) if len(args) > 1 else Path("/tmp/brand-run")
        cmd_smoke(outdir)

    elif cmd == "shot":
        brand = args[1] if len(args) > 1 else "all"
        # optional: "light" or "dark" as next arg, then outdir
        theme = "light"
        outdir = Path("/tmp/brand-run")
        if len(args) > 2 and args[2] in ("light", "dark"):
            theme = args[2]
            outdir = Path(args[3]) if len(args) > 3 else outdir
        elif len(args) > 2:
            outdir = Path(args[2])
        cmd_shot(brand, theme, outdir)

    elif cmd == "build":
        brand = args[1] if len(args) > 1 else "all"
        cmd_build(brand)

    else:
        print(f"Unknown command: {cmd}")
        print(__doc__)
        sys.exit(1)
